presto-parser/src/main/antlr4/com/facebook/presto/sql/parser/SqlBase.g4

```diff
@@ -283,6 +284,14 @@ primaryExpression
     | '(' expression ')'                                                             #parenthesizedExpression
     ;
 
+mapKey
```

Comment:
Nevermind, I figured it out. You can accumulate rules into labels that become fields in the parse tree node: https://github.com/antlr/antlr4/blob/master/doc/parser-rules.md.